server/handlers.js

```diff
@@ -110,19 +112,30 @@ const readHandler = async (req, res) => {
     const resData = []
     await client.connect()
     const db = client.db('pang');
-    try {
-        db.collection('users').findOne({_id: userId }, (err, result) => { 
-            console.log(result, 'result')
-            // const following = result.following;
-            // following.forEach(user => {
-            //     db.collection('posts').find({_id: user})
-            //     .toArray((err, result) => { 
-            //         result.forEach(post => {
-            //             resData.push(post)
-            //         }) 
-            //     })
-            // })
-            // res.status(200).json({status: 200, data: resData.sort((a, b) => b.date - a.date)})
+    try { 
+        await db.collection('users').findOne({_id: userId }, (err, result) => {
+            console.log(result.following)
+            console.log(result.following == [])
+            if(result.following.length) {
+                console.log('pangy')
+                const following = result.following;
+                following.forEach(user => {
+                    db.collection('posts').find({user_id: user})
+                    .toArray((err, result) => { 
```

Comment:
Generally you want to avoid making many queries to your DB for a single operation. This is costly in term of ressources, therefore it's generally best to be able to retrieve information in one query. 

In this case you should be able to pass an array of matching userId to get all the corresponding posts in one operation : https://stackoverflow.com/questions/32264225/how-to-get-multiple-document-using-array-of-mongodb-id/32264630

This would be true for several other queries in the handlers below